src/app/pages/MediaAssetPage/MediaAssetPage.jsx

```diff
@@ -52,10 +52,18 @@ const MediaAssetPage = ({ pageData }) => {
     ['relatedContent', 'groups', 0, 'promos'],
     pageData,
   );
-  const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
-  const indexImageLocator = indexImagePath
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
+      return null;
+    }
+
+    const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
```

Comment:
This is quite hard to read. If we check that indexImagePath exists before line 62, we can get rid of the checks on lines 62 & 63

Reply:
Could you suggest some pseudocode which would make it easier to read?

Reply:
Check if indexImagePath has a value. If it doesn't, return early. Then remove the `indexImagePath &&` and `imageParts &&` checks as they will no longer be needed